gen-diff/src/graph.rs

```diff
@@ -0,0 +1,607 @@
+use std::collections::{HashMap, HashSet};
+
+use gen_core::{
+    HashId, NO_CHROMOSOME_INDEX,
+    Strand::{self, Forward},
+    is_end_node, is_start_node, is_terminal,
+};
+use gen_graph::{GenGraph, GraphEdge, GraphNode};
+use gen_models::{
+    block_group_edge::BlockGroupEdge, changesets::ChangesetModels, edge::Edge, node::Node,
+    sequence::Sequence, session_operations::DependencyModels,
+};
+use itertools::Itertools;
+use petgraph::{Direction, graphmap::DiGraphMap};
+
+#[derive(Clone, Copy, Debug, Eq, PartialEq, Hash, Ord, PartialOrd)]
+pub struct DiffGraphNode {
+    pub node: GraphNode,
+    pub is_new: bool,
+}
+
+#[derive(Clone, Copy, Debug, Eq, PartialEq, Hash, Ord, PartialOrd)]
+pub struct DiffGraphEdge {
+    pub edge: GraphEdge,
+    pub is_new: bool,
+}
+
+pub type DiffGenGraph = DiGraphMap<DiffGraphNode, Vec<DiffGraphEdge>>;
+
+pub struct DiffGenGraphRef<'a>(pub &'a DiffGenGraph);
+
+impl<'a> From<&'a DiffGenGraph> for DiffGenGraphRef<'a> {
+    fn from(graph: &'a DiffGenGraph) -> Self {
+        Self(graph)
+    }
+}
+
+impl<'a> From<DiffGenGraphRef<'a>> for GenGraph {
+    fn from(val: DiffGenGraphRef<'a>) -> Self {
+        let mut graph = GenGraph::new();
+        for node in val.0.nodes() {
+            graph.add_node(node.node);
+        }
+        for (src, dest, edges) in val.0.all_edges() {
+            let mapped_edges = edges.iter().map(|edge| edge.edge).collect::<Vec<_>>();
+            graph.add_edge(src.node, dest.node, mapped_edges);
+        }
+        graph
+    }
+}
+
+impl From<DiffGraphNode> for GraphNode {
+    fn from(node: DiffGraphNode) -> Self {
+        node.node
+    }
+}
+
+impl From<DiffGraphEdge> for GraphEdge {
+    fn from(edge: DiffGraphEdge) -> Self {
+        edge.edge
+    }
+}
+
+pub fn get_diff_graph(
```

Comment:
this is more or less a copy/paste of `get_change_graph`, that function may go away in favor of this.